### ISSUE TITLE
CMake version checking does not check minimum required version

### ISSUE THREAD

**Issue by KrisThielemans** [gflags/gflags#358]
This code actually doesn't do any version checking
https://github.com/gflags/gflags/blob/a738fdf9338412f83ab3f26f31ac11ed3f3ec4bd/cmake/version.cmake.in#L13-L21
This causes upstream packages that require a certain gflag version to fail when the gflag version is older than what they need.

**Comment by schuhschuh** [comment]
You're right. The reason behind this is that accurate compatibility information between released gflags versions is unknown.

If `glog` requires a very specific version such as 2.2.2, they should use `find_package (gflags 2.2.2 EXACT)`.

> The EXACT option requests that the version be matched exactly. This option is incompatible with the specification of a version range.

https://cmake.org/cmake/help/latest/command/find_package.html
